src/dist/manifestation/tests.rs

```diff
@@ -31,6 +33,7 @@ use crate::{
 const SHA256_HASH_LEN: usize = 64;
 
 /// Temporary thunk to support asyncifying from underneath.
+#[deprecated]
```

Comment:
I think we should squash this commit and the next one; they're not logically independent (this just adds unused code which would trigger warnings when running `cargo check` at this commit) and it's easier to review the new code when it's in the same commit as the old code being removed.